src/System.Net.Http/src/System/Net/Http/Managed/HttpProxyConnectionHandler.cs

```diff
@@ -22,10 +22,9 @@ public HttpProxyConnectionHandler(HttpConnectionSettings settings, HttpMessageHa
         {
             Debug.Assert(innerHandler != null);
             Debug.Assert(settings._useProxy);
-            Debug.Assert(settings._proxy != null || s_proxyFromEnvironment.Value != null);
 
             _innerHandler = innerHandler;
-            _proxy = settings._proxy ?? new PassthroughWebProxy(s_proxyFromEnvironment.Value);
+            _proxy = settings._proxy ?? GetDefaultProxy();
```

Comment:
I think you want to use s_defaultProxy.Value here, otherwise you are going to re-read the env vars every time here.

Reply:
It depends whether we want to respect changes made via Environment.SetEnvironmentVariable.  If we do, we need to check each time.  If we don't, we should definitely just do the creation once and reuse it.

I don't have a strong preference either way, but I'm inclined to agree that we don't need to respect changes to it, and that these environment variables are about allowing the environment to be configured when the process starts rather than it being yet another mechanism for the process itself to influence itself (rather than just configuring the proxy appropriately on the instance it's creating).